src/aviary/env.py

```diff
@@ -410,9 +410,12 @@ def get_new_env_by_idx(self, idx: int) -> TEnvironment:
         )
 
     def get_new_env(self) -> TEnvironment:
-        """Get an env from a non-indexable dataset."""
+        """Get an env from a non-indexable dataset, or if there's just one item."""
+        if len(self) == 1:
+            return self.get_new_env_by_idx(0)
```

Comment:
IMO this is confusing - `get_new_env()` is used for sampling envs from non-indexable datasets. Now a user has to keep track of 3 scenarios:
- a dataset of size >1 will crash
- a dataset of size 1 will return the env
- a dataset of any size (even undefined) may override this and do a 3rd thing

I don't feel strongly about this since this method doesn't see heavy use, but not clear saving a few characters over `dataset.get_new_env_by_idx(0)` is worth this complexity.

Reply:
Oh another problem: if `__len__` is not defined, this will throw a TypeError, not a NotImplementedError.

Reply:
Thanks for sharing, nice thinking and agreed. Reverted this change